picky-asn1-x509/src/algorithm_identifier.rs

```diff
@@ -1083,4 +1172,22 @@ mod tests {
             concat!("deserialized ", stringify!($item), " doesn't match")
         );
     }
+
+    #[test]
+    fn raw_algorithm_roundtrip() {
+        let encoded = [
+            0x30, 0x0C, 0x06, 0x08, 0x2A, 0x86, 0x48, 0x86, 0xF7, 0x0D, 0x02, 0x09, 0x05, 0x00,
+        ];
+
+        let decoded: RawAlgorithmIdentifier =
+            picky_asn1_der::from_bytes(&encoded).expect("failed RawAlgorithmIdentifier deserialization");
+
+        let expected = RawAlgorithmIdentifier::from_parts(oids::hmac_with_sha256(), None);
+
+        pretty_assertions::assert_eq!(decoded, expected);
+
+        check_serde!(decoded: RawAlgorithmIdentifier in encoded);
+    }
+
+    // TODO PACMANCODER: Add test with parameters set to some value
```

Comment:
Nit: it’s frequent to attach TODO comment author this way:
```suggestion
    // TODO(pacmancoder): Add test with parameters set to some value
```